### ISSUE TITLE
Prometheus Exporter should have an option to export using OpenMetrics format

### ISSUE THREAD

**Issue by paivagustavo** [open-telemetry/opentelemetry-go#3468]
### Problem Statement
Prometheus exporter only expose metrics using the Prometheus format, which is a subset of the OpenMetrics format. Ideally, we should be able to specify an option to export metrics using the OpenMetrics format.

Given that we don't expose an OpenMetrics format, we lose information about metric's units, which make some required processing of Prometheus metric to OTLP [as the specs define](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#prometheus-metric-points-to-otlp) to be skipped, e..g, unit suffixes added by the otel-go are not able to be removed due to the lack of unit information as a limitation of the prometheus format.

### Proposed Solution
A `prometheus.WithOpenMetrics()` option to enable the OpenMetrics format, which should contains unit metadata and implement the [OTLP Metric points to Prometheus](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#otlp-metric-points-to-prometheus).
> The Unit of an OTLP metric point MUST be added as [OpenMetrics UNIT metadata](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#metricfamily). Additionally, the unit MUST be added as a suffix to the metric name, and SHOULD be converted to [base units](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#units-and-base-units) recommended by OpenMetrics when possible. The unit suffix comes before any type-specific suffixes.

#### Alternatives
Another alternative would always enable this behavior, but having an option to enable/disabled can be useful.

**Comment by MrAlias** [comment]
This doesn't look to be required for GA, moving to the post-GA project.

**Comment by MrAlias** [comment]
Note from SIG meeting: this will likely depend on the underlying Prometheus library supporting OpenMetrics before we want to also support it.

**Comment by dashpole** [comment]
You can use OpenMetrics today with the prometheus exporter.  In the example, you can change this line: https://github.com/open-telemetry/opentelemetry-go/blob/0e874c3468c2cbc5bbaf075537be6c0f2152b5ef/example/prometheus/main.go#L93

To:
```golang
http.Handle("/metrics", HandlerFor(prometheus.DefaultGatherer, HandlerOpts{EnableOpenMetrics: true})) 
```

**Comment by dashpole** [comment]
I don't think this blocks the prometheus exporter GA.  Adding UNIT metadata to the exporter after the go client has support can be done as an addition later.

**Comment by dashpole** [comment]
Blocked by https://github.com/prometheus/client_golang/issues/684
